File: python/problem1513.py
class Solution:
    def numSub(self, s: str) -> int:
        oneCount = 0
        result = 0
        for digit in s:
            if digit == '1':
                oneCount += 1
            else:
                result += (oneCount * (oneCount + 1)) // 2
                oneCount = 0
        result += (oneCount * (oneCount + 1)) // 2
        return result % 1000_000_007

File: Java/problem1513.java
class Solution {
    public int numSub(String s) {
        int MOD = 1000_000_007;
        int res = 0;
        int oneCount = 0;
        for (char c : s.toCharArray()) {
            if (c == '1') {
                ++oneCount;
            }
            else {
                res = (res + (int)(((long)oneCount * (long)(oneCount + 1)) % MOD) / 2) % MOD;
                oneCount = 0;
            }
        }
        res = (res + (int)(((long)oneCount * (long)(oneCount + 1)) % MOD) / 2) % MOD;
        return res;
    }
}
